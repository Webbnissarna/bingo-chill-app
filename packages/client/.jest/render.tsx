import React from "react";
import { render as rtlRender } from "@testing-library/react";

export function render(ui: React.ReactElement) {
  return rtlRender(<>{ui}</>);
}
