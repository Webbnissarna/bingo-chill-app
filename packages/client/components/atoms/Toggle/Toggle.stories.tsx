import { useState } from "react";
import Toggle from ".";
import type { StoryDefault } from "@ladle/react";

export function Default() {
  const [value, setValue] = useState(false);
  return (
    <div className="bg-polarNight-1 p-5">
      <Toggle checked={value} onChange={setValue} />
    </div>
  );
}

export default {
  title: "Atoms / Toggle",
} satisfies StoryDefault;
