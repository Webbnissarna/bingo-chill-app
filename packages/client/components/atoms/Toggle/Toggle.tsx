interface ToggleProps {
  checked: boolean;
  onChange: (newValue: boolean) => void;
}

export default function Toggle({
  checked,
  onChange,
}: ToggleProps): JSX.Element {
  return (
    <input
      className="w-7 h-7 outline-none border-none rounded bg-polarNight-0 text-frost-3 focus:border-none cursor-pointer active:text-frost-2 transition"
      type="checkbox"
      checked={checked}
      onChange={(e) => onChange(e.currentTarget.checked)}
    />
  );
}
