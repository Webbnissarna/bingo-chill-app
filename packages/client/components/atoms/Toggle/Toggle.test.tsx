import { render } from "../../../.jest/render";
import Toggle from ".";

describe("Toggle", () => {
  it("renders", () => {
    const { asFragment } = render(
      <Toggle checked onChange={() => undefined} />,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
