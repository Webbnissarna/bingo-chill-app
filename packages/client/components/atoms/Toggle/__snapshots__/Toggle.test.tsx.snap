// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle renders 1`] = `
<DocumentFragment>
  <input
    checked=""
    class="w-7 h-7 outline-none border-none rounded bg-polarNight-0 text-frost-3 focus:border-none cursor-pointer active:text-frost-2 transition"
    type="checkbox"
  />
</DocumentFragment>
`;
